lib/do_it_web/controllers/list_controller.ex

```diff
@@ -3,10 +3,19 @@ defmodule DoItWeb.ListController do
   alias DoIt.List
   alias DoIt.Repo
 
-  def show(conn, _params) do
-    list = %{title: "Foo"}
+  def show(conn, %{"id" => id}) do
+    case Repo.get(List, id) do
+      list = %List{} ->
```

Comment:
Watch out, pattern matching tries to match the right to the left. Here you want to check that `list` is a type of `%List{}`, not that `%List{}` is a type of `list`. 
```suggestion
      %List{} = list ->
```